cluster/points_writer.go

```diff
@@ -280,6 +282,7 @@ func (w *PointsWriter) writeToShard(shard *meta.ShardInfo, database, retentionPo
 			// If the write returned an error, continue to the next response
 			if err != nil {
 				w.Logger.Printf("write failed for shard %d on node %d: %v", shard.ID, nodeID, err)
+				writeError = err
```

Comment:
Since the system only supports sending one error back, I don't have a
strong preference.

That said if a write request has many individual point failures, perhaps it
would be easier dealing with the earlier ones first. Practically speaking I
will examine my multipoint request from first point to last. So first might
be better. Your call.

On Wednesday, June 10, 2015, Jason Wilder <notifications@github.com> wrote:

> In cluster/points_writer.go
> <https://github.com/influxdb/influxdb/pull/2882#discussion_r32163443>:
>
> > @@ -280,6 +282,7 @@ func (w *PointsWriter) writeToShard(shard *meta.ShardInfo, database, retentionPo
> >  			// If the write returned an error, continue to the next response
> >  			if err != nil {
> >  				w.Logger.Printf("write failed for shard %d on node %d: %v", shard.ID, nodeID, err)
> > +				writeError = err
>
> It can only return one error. Could make it first too. Any preference?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/2882/files#r32163443>.
>